compiler/rustc_const_eval/src/const_eval/valtrees.rs

```diff
@@ -98,6 +98,12 @@ fn const_to_valtree_inner<'tcx>(
             Ok(ty::ValTree::Leaf(val.assert_int()))
         }
 
+        ty::Pat(base, ..) => {
+            let mut place = place.clone();
+            place.layout = ecx.layout_of(*base).unwrap();
```

Comment:
Can you leave a comment explaining why this is so? It seems weird that we're setting it to the scrutinee's layout (i.e. w/o niches), unless you are just using this for the `Ty` part of `TyAndLayout`.